### PR TITLE
Inverted key/value for select options

### PR DESCRIPTION
**Pull request by ndbroadbent**
In response to #71

### CODE REVIEW

**Comment by mastilver** [comment]
Why not fixing the issue rather that inverting the docs? It's obviously a bug

Also it doesn't fix the issue for `Array`s:

```
select('test', [
  'item1',
  'item2',
  'item3'
], 'item1');
```

When the expected is:

```
select('test', [
  'item1',
  'item2',
  'item3'
], 0);
```

File: example/stories/index.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ stories.add('with all knobs', () => {
 
   const bold = boolean('Bold', false);
   const selectedColor = color('Color', 'black');
-  const favoriteNumber = number('Favorite Number', 42);
+  const favoriteNumber = select('Favorite Number', { 'Seven': 7, 'Forty-two': 42, 'Nine thousand': 9000 }, 42);
```
Comment (mthuret) [inline example/stories/index.js]: I think it would be better to add a new line, keeping the simple "number" example.

**Review by mthuret — CHANGES_REQUESTED** [review]

**Comment by mthuret** [comment]
@mastilver Can you explain what you've expected? It seems that this doesn't just invert the doc. 

I think this syntax is more natural. But this is a breaking change. I think it's ok to go for a major version but WDYT @mnmtanish and @apexskier?

**Comment by mastilver** [comment]
To me the docs sounds right, That's the way I would want it to work. Especially when you take array in consideration

Cc @lifehacket

**Comment by apexskier** [comment]
I actually very much like this. When I first used knobs, the original syntax was pretty surprising, but this definitely requires a major version bump.

**Comment by ndelangen** [comment]
Hey @ndbroadbent 

Thank you so much for this PR! We're moving over to a mono-repo, since that makes a lot of sense for a project like this.
We would ❤️ it if you could create a PR here: https://github.com/storybooks/storybook

That'd be awesome, thanks!

**Pull request closed without merging**